### DISCUSSION TITLE
Is there a way to disable caching?

### DISCUSSION THREAD

**Discussion by natefinch** [golangci/golangci-lint#3324] [Q&A] [answered]
I can tell golangci-lint is caching because it'll return WAY faster the second time it runs. But it doesn't seem to detect that I've changed the config file, and so it'll show errors for a linter that I just added to the disabled list, or otherwise quieted. It's quite annoying.  Sometimes it'll figure out it needs to rerun, but sometimes it won't, and I haven't figured out how to get it to not cache.  I would hope it would at least hash the config file and know that if the config file changes, it should ignore the cache... but it doesn't seem to.

Any help? I'd love a --ignore-cache or something similar, but I didn't see anything in the command flags.

**Comment by ldez** [comment, marked as answer]
Hello,

the cache is the "secret" trick to improve performance, so there is no option to disable it.

But you can clear the cache:
```
golangci-lint cache clean
```

**Reply by natefinch** [reply]
I think I'll file a bug, since there seems to be something wonky with the cache, such that it's not picking up changes I make to the configuration.  but thanks for the tip on clearing the cache.

**Comment by flimzy** [comment]
I wouuld love a way to disable the cache for a single run, without deleting the cache.

It's useful when profiling different configurations.
